tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSFQQNConsumerTest.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.activemq.artemis.utils.RandomUtil;
+import org.apache.activemq.artemis.utils.Wait;
+import org.junit.Test;
+
+public class JMSFQQNConsumerTest extends MultiprotocolJMSClientTestSupport {
+
+   @Test
+   public void testFQQNTopicConsumerWithSelectorAMQP() throws Exception {
+      testFQQNTopicConsumerWithSelector(AMQPConnection);
+   }
+
+   @Test
+   public void testFQQNTopicConsumerWithSelectorOpenWire() throws Exception {
+      testFQQNTopicConsumerWithSelector(OpenWireConnection);
+   }
+
+   @Test
+   public void testFQQNTopicConsumerWithSelectorCore() throws Exception {
+      testFQQNTopicConsumerWithSelector(CoreConnection);
+   }
+
+   private void testFQQNTopicConsumerWithSelector(ConnectionSupplier supplier) throws Exception {
+      final String queue = "queue";
+      final String address = "address";
+      final String filter = "prop='value'";
+      try (Connection c = supplier.createConnection()) {
+         Session s = c.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic t = s.createTopic(CompositeAddress.toFullyQualified(address, queue));
+         MessageConsumer mc = s.createConsumer(t, filter);
+         Wait.assertTrue(() -> server.locateQueue(queue) != null, 2000, 100);
+         assertNotNull(server.locateQueue(queue).getFilter());
+         assertEquals(filter, server.locateQueue(queue).getFilter().getFilterString().toString());
+      }
+   }
+
+   @Test
+   public void testFQQNQueueConsumerWithSelectorAMQP() throws Exception {
+      testFQQNQueueConsumerWithSelector(AMQPConnection);
+   }
+
+   @Test
+   public void testFQQNQueueConsumerWithSelectorOpenWire() throws Exception {
+      testFQQNQueueConsumerWithSelector(OpenWireConnection);
+   }
+
+   @Test
+   public void testFQQNQueueConsumerWithSelectorCore() throws Exception {
+      testFQQNQueueConsumerWithSelector(CoreConnection);
+   }
+
+   private void testFQQNQueueConsumerWithSelector(ConnectionSupplier supplier) throws Exception {
+      final String queue = "queue";
+      final String address = "address";
+      final String prop = "prop";
+      final String value = RandomUtil.randomString();
+      final String filter = prop + "='" + value + "'";
+      try (Connection c = supplier.createConnection()) {
+         Session s = c.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Queue q = s.createQueue(CompositeAddress.toFullyQualified(address, queue));
+         MessageConsumer mc = s.createConsumer(q, filter);
+         Wait.assertTrue(() -> server.locateQueue(queue) != null, 2000, 100);
+         assertNull(server.locateQueue(queue).getFilter());
+         MessageProducer p = s.createProducer(q);
+         Message m = s.createMessage();
+         m.setStringProperty(prop, value);
+         p.send(m);
+         c.start();
+         assertNotNull(mc.receive(1000));
+         m = s.createMessage();
+         m.setStringProperty(prop, RandomUtil.randomString());
+         assertNull(mc.receive(1000));
```

Comment:
The second creation+prop-set for 'm' seems unused since all it does after is cehck nothing is received.

Some separation would help readability, perhaps make things like that more obvious.

Reply:
I personally would put this to a loop and sent several messages.